src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSourceWriter.java

```diff
@@ -5,23 +5,24 @@
  */
 package com.microsoft.spring.data.gremlin.conversion;
 
+import com.microsoft.spring.data.gremlin.repository.support.GremlinEntityInformation;
 import org.springframework.lang.NonNull;
 import org.springframework.util.Assert;
 
 public class BasicGremlinSourceWriter {
 
     private GremlinEntityInformation entityInformation;
 
-    public BasicGremlinSourceWriter(@NonNull Object domain) {
-        this.entityInformation = new GremlinEntityInformation(domain);
+    public BasicGremlinSourceWriter(@NonNull Class<?> domainClass) {
+        this.entityInformation = new GremlinEntityInformation(domainClass);
     }
 
     protected String getPersistentEntityId() {
-        return entityInfo.getIdField().toString();
+        return this.entityInformation.getIdField().toString();
```

Comment:
what if not specific id field?

Reply:
GremlinEntityInformation will take care of domainClass in constructor, if we cannot figure out the only one id Field. One runtime exception will be raised.